### PR TITLE
Server: remove some redundant code

### LINKED ISSUE #8461: Set a timeout for proxy init
## Enhancement
https://github.com/pingcap/tiflash/blob/68181844e5a9e0643583bd51ddcf03e909d71cdb/dbms/src/Server/Server.cpp#L1041-L1042

now TiFlash server will keep on waiting for proxy init, we should set up a timeout for better detect error.

### PR DESCRIPTION
**Pull request by Lloyd-Pottiger**
### What problem does this PR solve?

Issue Number: ref #6233

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Server/Server.cpp [hunk 1]
```diff
@@ -1036,18 +1036,21 @@ int Server::main(const std::vector<std::string> & /*args*/)
     if (proxy_conf.is_proxy_runnable)
     {
         proxy_runner.run();
-
+        // wait for proxy initializing, timeout is 60s
+        static const UInt64 PROXY_START_TIMEOUT_SECONDS = 60;
```
[the remark this answers is not in the document]
Comment (Lloyd-Pottiger) [reply dbms/src/Server/Server.cpp:1040]: https://github.com/pingcap/tidb-engine-ext/blob/24c3802ffee888a64f82d8b477da53c367eee88b/proxy_components/proxy_server/src/run.rs#L125-L176 

seems no time cost operation during init.

File: dbms/src/Server/Server.cpp [hunk 2]
```diff
@@ -1036,18 +1036,21 @@ int Server::main(const std::vector<std::string> & /*args*/)
...
         while (!tiflash_instance_wrap.proxy_helper)
-            std::this_thread::sleep_for(std::chrono::milliseconds(200));
-        LOG_INFO(log, "tiflash proxy is initialized");
-        if (tiflash_instance_wrap.proxy_helper->checkEncryptionEnabled())
         {
-            auto method = tiflash_instance_wrap.proxy_helper->getEncryptionMethod();
-            LOG_INFO(log, "encryption is enabled, method is {}", IntoEncryptionMethodName(method));
+            if (watch.elapsedSeconds() > PROXY_START_TIMEOUT_SECONDS)
+            {
+                proxy_runner.join();
```
[the remark this answers is not in the document]
Comment (Lloyd-Pottiger) [reply dbms/src/Server/Server.cpp]: fixed in https://github.com/pingcap/tiflash/pull/8462/commits/9c2763981f49254c31759a152acfeffac9b54f2d

File: dbms/src/Server/Server.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (CalvinNeo) [reply dbms/src/Server/Server.cpp:1048]: It is not enough, in https://github.com/pingcap/tidb-engine-ext/blob/24c3802ffee888a64f82d8b477da53c367eee88b/proxy_components/proxy_ffi/src/raftstore_proxy.rs#L207, we will wait for at most 60s until cluster boostrapped, this time could be modified longer if TiKV spent more time booting clusters.

[the remark this answers is not in the document]
Comment (CalvinNeo) [reply dbms/src/Server/Server.cpp:1048]: I think we'd better make it a larger value, before we make the `RETRY_LIMIT` a configurable value(is it OK to have a config for this?) to be read from proxy's config by FFI.

[the remark this answers is not in the document]
Comment (Lloyd-Pottiger) [reply dbms/src/Server/Server.cpp]: Seems not a good idea, revert.

File: dbms/src/Server/Server.cpp [hunk 3]
```diff
@@ -1032,22 +1032,33 @@ int Server::main(const std::vector<std::string> & /*args*/)
...
-        LOG_INFO(log, "tiflash proxy is initialized");
-        if (tiflash_instance_wrap.proxy_helper->checkEncryptionEnabled())
         {
-            auto method = tiflash_instance_wrap.proxy_helper->getEncryptionMethod();
-            LOG_INFO(log, "encryption is enabled, method is {}", IntoEncryptionMethodName(method));
+            if (watch.elapsedSeconds() > PROXY_START_TIMEOUT_SECONDS)
+            {
+                shutdown_proxy();
+                LOG_ERROR(log, "wait for tiflash proxy initializing timeout");
+                throw Exception("wait for tiflash proxy initializing timeout");
```
[the remark this answers is not in the document]
Comment (Lloyd-Pottiger) [reply dbms/src/Server/Server.cpp]: I found even proxy version is not matched with tiflash, tiflash will not exit.

File: dbms/src/Server/Server.cpp [hunk 4]
```diff
@@ -1036,18 +1036,10 @@ int Server::main(const std::vector<std::string> & /*args*/)
...
         proxy_runner.run();
-
         LOG_INFO(log, "wait for tiflash proxy initializing");
         while (!tiflash_instance_wrap.proxy_helper)
             std::this_thread::sleep_for(std::chrono::milliseconds(200));
         LOG_INFO(log, "tiflash proxy is initialized");
-        if (tiflash_instance_wrap.proxy_helper->checkEncryptionEnabled())
-        {
-            auto method = tiflash_instance_wrap.proxy_helper->getEncryptionMethod();
-            LOG_INFO(log, "encryption is enabled, method is {}", IntoEncryptionMethodName(method));
```
Comment (CalvinNeo) [inline dbms/src/Server/Server.cpp:1047]: why this code is obsolete?

Comment (Lloyd-Pottiger) [reply dbms/src/Server/Server.cpp:1047]: We have done this below.

**Comment by Lloyd-Pottiger** [comment]
Close this, will open another pr if we find a better way to slove https://github.com/pingcap/tiflash/issues/8461

**Pull request closed without merging**